test/src/ui/pages/registration/registration_notifier_test.dart

```diff
@@ -21,31 +22,111 @@ void main() {
   setUp(() {
     mockAuthenticationRepository = MockAuthenticationRepository();
     mockAuthenticationMessageProvider = MockAuthenticationMessageProvider();
+    mockFieldMessageProvider = MockFieldMessageProvider();
     registrationNotifier = RegistrationNotifier(
       mockAuthenticationRepository,
       mockAuthenticationMessageProvider,
       mockFieldMessageProvider,
     );
-
-    registrationNotifier.updatePassword("mock");
-    registrationNotifier.updateEmail("mock");
-    registrationNotifier.updateName("mock");
   });
 
   group('notifier tests', () {
-    test('should emit loaded registration state when registering new user', () {
-      when(mockAuthenticationRepository.createUser(any, any)).thenAnswer((_) async => Future.value());
+    test('should set loaded registration state', () {
+      when(mockAuthenticationRepository.createUser(any, any))
+          .thenAnswer((_) async => Future.value());
       when(mockAuthenticationRepository.updateUser(any)).thenAnswer((_) async => Future.value());
 
+      registrationNotifier.updatePassword("mock");
+      registrationNotifier.updateEmail("mock");
+      registrationNotifier.updateName("mock");
+
+      registrationNotifier.verifyStateInOrder(
+        testFunction: registrationNotifier.validateFieldsAndCreateUser,
+        matchersMethods: [
+          () {
+            expect(registrationNotifier.state.isLoading, true);
+          },
+          () {
+            verifyInOrder([
+              mockAuthenticationRepository.createUser(any, any),
+              mockAuthenticationRepository.updateUser(any)
+            ]);
+            expect(registrationNotifier.state.isLoading, false);
+          }
+        ],
+      );
+    });
+
+    test('should set field error registration state', () {
+      const errorMessage = "blad";
+      when(mockFieldMessageProvider.getMessage(any)).thenReturn(errorMessage);
+
+      registrationNotifier.verifyStateInOrder(
+        testFunction: registrationNotifier.validateFieldsAndCreateUser,
+        matchersMethods: [
+          () {
+            expect(registrationNotifier.state.isLoading, true);
+          },
+          () {
+            verifyZeroInteractions(mockAuthenticationRepository);
+            expect(registrationNotifier.state.isLoading, false);
+            expect(registrationNotifier.state.passwordError, errorMessage);
+            expect(registrationNotifier.state.emailError, errorMessage);
+            expect(registrationNotifier.state.nameError, errorMessage);
+          }
+        ],
+      );
+    });
+
+    test('should return error when creating user', () {
+      const errorMessage = "blad";
+      registrationNotifier.updatePassword("mock");
```

Comment:
dałbym mimo wszystko różne nazwy, żeby później zrobić asercję czy rzeczywiście to co idzie do repo zgadza sie z tym co updatował widok